lib/EnsEMBL/REST/Controller/VEP.pm

```diff
@@ -438,9 +449,95 @@ sub _include_user_params {
   $vep_params{$_} = $c->stash->{$_} for qw(va vfa svfa tva csa sa ga), map {$_.'_adaptor'} @REG_FEAT_TYPES;
   if (!$c->stash->{'RegulatoryFeature_adaptor'}) { delete $vep_params{regulatory}; };
 
+  my $plugin_config = $self->_configure_plugins($c,$user_config,\%vep_params);
+  $vep_params{plugins} = $plugin_config if $plugin_config;
+
   # $c->log->debug("After ".Dumper \%vep_params);
   return \%vep_params;
 }
 
+sub _configure_plugins {
+  my ($self,$c,$user_config,$vep_config) = @_;
+
+  # backup up/down distances
+  $UPSTREAM_DISTANCE_BAK   = $Bio::EnsEMBL::Variation::Utils::VariationEffect::UPSTREAM_DISTANCE;
+  $DOWNSTREAM_DISTANCE_BAK = $Bio::EnsEMBL::Variation::Utils::VariationEffect::DOWNSTREAM_DISTANCE;
+
+  # add dir_plugins to Perl's list of include dirs
+  # otherwise the plugins have to be somewhere in PERL5LIB on startup
+  unshift @INC, $vep_config->{dir_plugins};
+  my @plugin_config = ();
+
+  # get config from file
+  my $plugin_config_file = $vep_config->{plugin_config};
+  return unless $plugin_config_file && -e $plugin_config_file;
+
+  open IN, $plugin_config_file or return;
+  my @content = <IN>;
+  close IN;
+  
+  my $VEP_PLUGIN_CONFIG = eval join('', @content);
+  $c->log->warn("Could not eval VEP plugin config file: $@\n") if $@;
```

Comment:
Would add here "and return if ..." to avoid the evaluation of the following statements.